### PR TITLE
[Feature][Branch][Rating] Refactoring and add README file

### PR DESCRIPTION
**Pull request by devethan**
## Description

Updated [Rating] component and add README file

## Related Issues

#53 

## Tests

Not yet

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui-native/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/components/shared/Rating/index.tsx [hunk 1]
```diff
@@ -59,76 +41,45 @@ const StyledImage = styled.Image`
...
-    const result = stars.fill(false);
-    let i = 0;
-    while (i < value) {
-      result[i] = true;
-      i++;
-    }
-
-    setStars(result);
+  const [position, setPosition] = useState(0);
+  const initStars = new Array(props.total).fill(false);
```
Comment (JeffGuKang) [inline src/components/shared/Rating/index.tsx]: How about to use `useMemo` for performance improvement?

**Review by JeffGuKang — CHANGES_REQUESTED** [review]
Normally, `onPress: () => void` or `onPress: (value: index) => void` styles are preferred when I define function parameter in interface instead of `Function`. It can be used as `onPress(): void` also. 

Ref: https://medium.com/@jeffbutsch/typescript-interface-functions-c691a108e3f1

File: src/components/shared/Rating/index.tsx (hunk 1 above)
Comment (devethan) [reply src/components/shared/Rating/index.tsx]: OK. I'll try it

File: src/components/shared/Rating/index.tsx [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-};
+interface OnchangeProps {
+  value: number;
+}
 
 interface Props {
   total: number;
   value: number;
-  onChange?: Function;
+  onChange?: (props: OnchangeProps) => void;
```
Comment (JeffGuKang) [inline src/components/shared/Rating/index.tsx]: I think `(value: number) => void` is enough for type.

File: src/components/shared/Rating/index.tsx [hunk 3]
```diff
@@ -59,80 +45,46 @@ const StyledImage = styled.Image`
...
-    while (i < value) {
-      result[i] = true;
-      i++;
-    }
-
-    setStars(result);
-
-    props.onChange && props.onChange({ value });
+  const _handlePress = (position: number): void => {
+    props.onChange && props.onChange({ value: position + 1 });
```
Comment (JeffGuKang) [inline src/components/shared/Rating/index.tsx]: If you change your `onChange` prop to `onChnage: (value: number) => void`, 
`onChange(position + 1)` will be match your type predifined.

**Review by JeffGuKang — CHANGES_REQUESTED** [review]
Almost done! Cheer up.

**Comment by JeffGuKang** [comment]
@devohno Pass the test by updating snapshot.

**Review by JeffGuKang — APPROVED** [review]

**Pull request merged**